app/models/ssh_execution_provider.rb

```diff
@@ -25,6 +25,24 @@ def ssh_key_passphrase(host)
       host_setting(host, :remote_execution_ssh_key_passphrase)
     end
 
+    def ssh_params(host)
+      proxy_selector = ::RemoteExecutionProxySelector.new
+      proxy = proxy_selector.determine_proxy(host, 'SSH')
+      {
+        :hostname => find_ip_or_hostname(host),
+        :proxy => proxy.class == Symbol ? proxy : proxy.url,
+        :ssh_user => ssh_user(host),
+        :ssh_port => ssh_port(host),
+        :ssh_password => ssh_password(host),
+        :ssh_key_passphrase => ssh_key_passphrase(host),
+        :ssh_key_file => File.expand_path(ForemanRemoteExecutionCore.settings.fetch(:ssh_identity_key_file))
+      }
+    end
+
+    def cockpit_url_for_host(host)
+      Setting[:remote_execution_cockpit_url] % { :host => host } if Setting[:remote_execution_cockpit_url].present?
```

Comment:
This might raise an exception and we should catch that, I'd say.

Reply:
what exception we are talking about?

Reply:
When `Setting[...]` is malformed for the `%` operator, such as with a typo like `"%{hos}"`